### PR TITLE
EVG-14227 fetch all-configs from github

### PR DESCRIPTION
**Pull request by ablack12**
[EVG-14227](https://jira.mongodb.org/browse/EVG-14227)

### Description 
So my previous attempt to test yamls using all-configs with yaml v3 didn't catch the problems because we were using the project that we'd previously parsed and stored in the DB. Instead, we need to fetch the config directly from github so it's untouched.

This will result in a lot of API calls so it might be expensive; I'm going to call it at kind of an off time I think.

### Testing 
Tested via staging.

### CODE REVIEW

File: operations/admin_all_configs.go [hunk 1]
```diff
@@ -54,38 +55,58 @@ func fetchAllProjectConfigs() cli.Command {
...
+			continue
+		}
+		grip.Infof("Downloading configuration for %s", p.Identifier)
+		versions, err := c.GetRecentVersions(p.Id)
+		if err != nil {
+			catcher.Wrapf(err, "failed to fetch recent versions for %s", p.Identifier)
+			continue
+		}
+		if len(versions) == 0 {
+			catcher.Errorf("WARNING: project %s has no versions", p.Identifier)
```
Comment (malikchaya2) [inline operations/admin_all_configs.go:95]: nit: %s --> '%s' can make it more readable

File: operations/admin_all_configs.go [hunk 2]
```diff
@@ -54,38 +55,58 @@ func fetchAllProjectConfigs() cli.Command {
...
+			continue
+		}
 
-	config, err := c.GetConfig(versions[0], useLegacy)
-	if err != nil {
-		return errors.Wrapf(err, "failed to fetch config for project %s, version %s", project, versions[0])
-	}
+		config, err := c.GetConfig(versions[0], useLegacy)
+		if err != nil {
+			catcher.Wrapf(err, "failed to fetch config for project %s, version %s", p.Identifier, versions[0])
```
Comment (malikchaya2) [inline operations/admin_all_configs.go:101]: same nit

File: operations/admin_all_configs.go [hunk 3]
```diff
@@ -54,38 +55,58 @@ func fetchAllProjectConfigs() cli.Command {
...
+			ConfigFile: p.RemotePath,
+		}
+		if exists := configDownloaded[repo]; exists {
+			grip.Infof("Configuration for project %s already downloaded", p.Identifier)
+			continue
+		}
+		grip.Infof("Downloading configuration for %s", p.Identifier)
+		versions, err := c.GetRecentVersions(p.Id)
+		if err != nil {
+			catcher.Wrapf(err, "failed to fetch recent versions for %s", p.Identifier)
```
Comment (malikchaya2) [inline operations/admin_all_configs.go:91]: same nit

File: operations/admin_all_configs.go [hunk 4]
```diff
@@ -54,38 +55,58 @@ func fetchAllProjectConfigs() cli.Command {
...
+			continue
+		}
+		configDownloaded[repo] = true
 
-	err = ioutil.WriteFile(project+".yml", config, 0644)
-	if err != nil {
-		return errors.Wrapf(err, "failed to write configuration for project %s", project)
+		err = ioutil.WriteFile(p.Identifier+".yml", config, 0644)
+		if err != nil {
+			catcher.Wrapf(err, "failed to write configuration for project %s", p.Identifier)
```
Comment (malikchaya2) [inline operations/admin_all_configs.go:108]: same nit

**Review by malikchaya2 — APPROVED** [review]
Approved, nits are only suggestions in case you want to make that change.

**Comment by ybrill** [comment]
could we allow it to take a developer's individual github token?

**Comment by ablack12** [comment]
> could we allow it to take a developer's individual github token?

Would this restrict us to downloading only the yamls that the user has access to? I don't have access to every repo, but I do need to look at all yamls. This is a feature we could add in the future if anyone has a use case, but I think it's pretty specific.

**Comment by ybrill** [comment]
I was thinking of it as a way around the issue you pointed out in the description

>This will result in a lot of API calls so it might be expensive

Though leaving it out works too!

**Pull request merged**